src/games/strategy/triplea/ui/screen/UnitsDrawer.java

```diff
@@ -24,86 +25,102 @@
+  private final static List<String> STATIC_UNITS = Arrays.asList("aaGun","factory");//TODO add every "static" non-movable unit type of every map.
+  private static byte drawUnitNationMode = 2;
   public UnitsDrawer(final int count, final String unitType, final String playerName, final Point placementPoint,
       final int damaged, final int bombingUnitDamage, final boolean disabled, final boolean overflow,
       final String territoryName, final IUIContext uiContext2) {
-    m_count = count;
-    m_unitType = unitType;
-    m_playerName = playerName;
-    m_placementPoint = placementPoint;
-    m_damaged = damaged;
-    m_bombingUnitDamage = bombingUnitDamage;
-    m_disabled = disabled;
-    m_overflow = overflow;
-    m_territoryName = territoryName;
-    m_uiContext = uiContext2;
+    this.count = count;
+    this.unitType = unitType;
+    this.playerName = playerName;
+    this.placementPoint = placementPoint;
+    this.damaged = damaged;
+    this.bombingUnitDamage = bombingUnitDamage;
+    this.disabled = disabled;
+    this.overflow = overflow;
+    this.territoryName = territoryName;
+    this.uiContext = uiContext2;
   }
 
-  public void prepare() {}
-
   public Point getPlacementPoint() {
-    return m_placementPoint;
+    return placementPoint;
   }
 
   public String getPlayer() {
-    return m_playerName;
+    return playerName;
   }
 
   @Override
   public void draw(final Rectangle bounds, final GameData data, final Graphics2D graphics, final MapData mapData,
       final AffineTransform unscaled, final AffineTransform scaled) {
-    if (m_overflow) {
+    
+    //If there are too many Units at one point a black line is drawn to make clear which units belong to where
+    if (overflow) {
       graphics.setColor(Color.BLACK);
-      graphics.fillRect(m_placementPoint.x - bounds.x - 2,
-          m_placementPoint.y - bounds.y + m_uiContext.getUnitImageFactory().getUnitImageHeight(),
-          m_uiContext.getUnitImageFactory().getUnitImageWidth() + 2, 3);
+      graphics.fillRect(placementPoint.x - bounds.x - 2,
+          placementPoint.y - bounds.y + uiContext.getUnitImageFactory().getUnitImageHeight(),
+          uiContext.getUnitImageFactory().getUnitImageWidth() + 2, 3);
     }
-    final UnitType type = data.getUnitTypeList().getUnitType(m_unitType);
+    final UnitType type = data.getUnitTypeList().getUnitType(unitType);
     if (type == null) {
-      throw new IllegalStateException("Type not found:" + m_unitType);
+      throw new IllegalStateException("Type not found:" + unitType);
+    }
+    final PlayerID owner = data.getPlayerList().getPlayerID(playerName);
+    final Image img = uiContext.getUnitImageFactory().getImage(type, owner, data, damaged > 0 || bombingUnitDamage > 0, disabled);
+    //If Mode 2 is selected the nation Flag gets drawn below the Unit
+    if(!STATIC_UNITS.contains(type.getName()) && drawUnitNationMode == 2){
+      final Image flag = uiContext.getFlagImageFactory().getFlag(owner);
```

Comment:
Yeah, I perhaps should have mentioned the duplication. IMO it is not too bad since it is one liners, and it makes the ordering switch very clear, for example:
```
switch( .. ) { 
  case drawBefore:
       drawLargeUnitFlagImage();
       graphics.drawImage(img, placementPoint.x - bounds.x, placementPoint.y - bounds.y, null);;
 case drawAfter:
       graphics.drawImage(img, placementPoint.x - bounds.x, placementPoint.y - bounds.y, null);;
       drawSmallUnitFlagImage();
}  
```
IMO the extra line of duplicated code is nicer here compared to logic nested within conditionals. The conditionals require some extra effort:
- need to understand what the 4 lines of code are doing in each conditional, how they are different, and figure out why exactly that is more important. Hiding that info behind a helper function and switch statement does not necessarily make it simpler, but it does accomplish some information hiding, so a person can figure out the overall flow very easily, and can dig into the details if they want to (rather than just being forced to dig into the details).

Regardless of using the switch statement, the logic within the `if` statements would do well to be extracted to helper functions, makes the differences between the two cleaner, and allows for this method to become slightly simpler.